authentication_service/models/auth.go

```diff
@@ -0,0 +1,24 @@
+package models
+
+import (
+	"time"
+)
+
+//User is the model that governs all notes objects retrived or inserted into the DB
+type User struct {
+	First_name    *string            `json:"first_name" validate:"required,min=2,max=100"`
+	Last_name     *string            `json:"last_name" validate:"required,min=2,max=100"`
+	Password      *string            `json:"Password" validate:"required,min=6""`
+	Email         *string            `json:"email" validate:"email,required"`
+	Token         *string            `json:"token"`
+	User_type     *string            `json:"user_type" validate:"required,eq=ADMIN|eq=USER""`
```

Comment:
also, the validator checks if it is USER | ADMIN, which suggests that this should be an `enum` using `iota` (go doesn't have real enums), like [so](https://yourbasic.org/golang/iota/)